amethyst_renderer/src/lib.rs

```diff
@@ -0,0 +1,285 @@
+
+        {
+            let encoders = self.encoders.as_mut_slice();
+            self.pool.install(|| {
+                pipe.stages()
+                    .par_iter()
+                    .zip(encoders)
+                    .filter(|&(stage, _)| stage.is_enabled())
+                    .for_each(|(stage, enc)| stage.apply(enc, scene));
+            });
+        }
+
+        for enc in self.encoders.iter_mut() {
+            enc.flush(&mut self.device);
+        }
+
+        self.device.cleanup();
+        #[cfg(feature = "opengl")]
+        self.window.swap_buffers().expect("OpenGL context has been lost");
+    }
+
+    /// Returns an immutable reference to the renderer window.
+    pub fn window(&self) -> &winit::Window {
+        self.window.as_winit_window()
+    }
+}
+
+impl Drop for Renderer {
+    fn drop(&mut self) {
+        use gfx::Device;
+        self.device.cleanup();
+    }
+}
+
+/// Represents a graphics backend for the renderer.
+struct Backend(pub types::Device, pub Factory, pub Target, pub Window);
+
+/// Creates the Direct3D 11 backend.
+#[cfg(all(feature = "d3d11", target_os = "windows"))]
+fn init_backend(wb: WindowBuilder, el: &EventsLoop) -> Result<Backend> {
+    use gfx_window_dxgi as win;
+
+    let (win, dev, mut fac, color) = win::init::<ColorFormat>(wb, el).unwrap();
+    let dev = gfx_device_dx11::Deferred::from(dev);
+
+    let size = win.get_inner_size_points().ok_or(Error::WindowDestroyed)?;
+    let (w, h) = (size.0 as u16, size.1 as u16);
+    let depth = fac.create_depth_stencil_view_only::<DepthFormat>(w, h)?;
+    let main_target = Target::from((
+        vec![ColorBuffer {
+            as_input: None,
+            as_output: color,
+        }],
+        DepthBuffer {
+            as_input: None,
+            as_output: depth,
+        },
+        size));
+
+    Ok(Backend(dev, fac, main_target, win))
+}
+
+#[cfg(all(feature = "metal", target_os = "macos"))]
+fn init_backend(wb: WindowBuilder, el: &EventsLoop) -> Result<Backend> {
+    use gfx_window_metal as win;
+
+    let (win, dev, mut fac, color) = win::init::<ColorFormat>(wb, el).unwrap();
+
+    let size = win.get_inner_size_points().ok_or(Error::WindowDestroyed)?;
+    let (w, h) = (size.0 as u16, size.1 as u16);
+    let depth = fac.create_depth_stencil_view_only::<DepthFormat>(w, h)?;
+    let main_target = Target::from((
+        vec![ColorBuffer {
+            as_input: None,
+            as_output: color,
+        }],
+        DepthBuffer {
+            as_input: None,
+            as_output: depth,
+        },
+        size));
+
+    Ok(Backend(dev, fac, main_target, win))
+}
+
+/// Creates the OpenGL backend.
+#[cfg(feature = "opengl")]
+fn init_backend(wb: WindowBuilder, el: &EventsLoop) -> Result<Backend> {
+    use glutin::{GlProfile, GlRequest};
+    use gfx_window_glutin as win;
+
+    let wb = glutin::WindowBuilder::from_winit_builder(wb)
+        .with_gl_profile(GlProfile::Core)
+        .with_gl(GlRequest::Latest)
```

Comment:
Could we have a temporary workaround for https://github.com/tomaka/glutin/issues/861 here?

Maybe something like this:

```rust
    .with_pixel_format(24, 8)
    .with_gl(glutin::GlRequest::GlThenGles {
        opengles_version: (2, 0),
        opengl_version: (2, 1),
})
```